### DISCUSSION TITLE
ASMB and Shimmer missing after v1.5

### DISCUSSION THREAD

**Discussion by Bg406** [iotaledger/firefly#2972] [Help & Support] [answered]
After I updated Firefly to v1.5 my iota balance is correct but my ASMB and Shimmer rewards from the last 90 day period are gone, though I can find them by my address on thetangle.org. Is there a way to recover those tokens to my wallet?

**Comment by MiKeF05** [comment]
Do you have more than 1 wallet inside the profile? The rewards are split into the different wallets. You can switch between them with the topbar.

**Reply by Bg406** [reply]
I just added a second wallet this morning, but neither show the staking rewards. Is it just a matter of running Balance Finder until my mouse breaks?

**Comment by Bg406** [comment, marked as answer]
Problem solved. For anyone else with this issue, a VPN (especially a hotel’s WiFi) will prevent the transfer of ASMB and Shimmer to your wallet.

**Comment by Alfext** [comment]
I had the same problem, and i've been finally able to solve It.
The key for the solution was this answer I received from other user, MikeF05.
"The most likely reason that your SMR and ASMB balance is not displayed is that you can't connect to github. The necessary info is fetched from github. You can check your staking address on [thetangle.org](http://thetangle.org/). You should see your rewards from the first staking period. There will be an alternative way to fetch the info in the next release."
Asking other people smarter than me, we have noticed that in my case, the problem was caused by a filter in the "host" file of my Windows.
Checking that file, we could see that the acces from my computer to the adress "[raw.githubusercontent.com](http://raw.githubusercontent.com/)" was blocked.
As soon as I eliminated that block in my host file, my ASMB and SMR reappeared in my FireFly.
This block filter could exist not only in the host file of a Windows O.S., but also in an adblock that anyone could have installed, or could even be blocked on a DNS level.

Anyone with this issue should try to ping to raw.githubusercontent.com in the first place, and verify if they are or aren´t getting an answer from that address. If not, they should verify what coould be blocking their access to that link, an solve it. The most likely is that solving that block, the ASMB and SMR reapear in their FireFly wallet.

**Reply by Mcblazeallot** [reply]
Im going to assume because I have windows that my problem is caused by not being able to connect to Github as well. But did your firefly wallet even give you the option to log in, or did it want you to make a new profile?

**Comment by Pedro1111111** [comment]
no me salen mis shimer en la tercera cuenta de firefly

**Comment by Mathias2679** [comment]
I have the same problem. I have 2 Wallets in my Firefly Wallet. In both i paticipate on staking. One Wallet shows me the tokens of the first staking period and the other doesn't :-( 
On thetangle.org i checked my adresses. The same here: correct entries for the first wallet, no rewards for the other.
Within the staking period i checked it often and everything was ok. After updating to 1.5 the rewards were disappeared.
Hope you can help me!!! Thank you

**Reply by MiKeF05** [reply]
Have you tried to use the balance finder in the settings a few times in a row? And you are sure you checked the correct address on thetangle.org with which you have staked?

**Reply by Mathias2679** [reply]
Hey, thanks for your answer!!
Yes I tried it already.
Your second question is a good idea. I checked every address in the address history. The address with i have staked is not in the history. It seems that the history is not complete. How can i restore the address history?

**Reply by MiKeF05** [reply]
Can you click in Firefly on the three dots in the wallet tab and then choose address history? There you should find all of your addresses from the wallet.

**Reply by Mathias2679** [reply]
Yes, there are the three dots. There are 4 addresses. I also checked them with thetangle.org
Result:
2 Addresses have not result
1 Address shows me a tranfer from 22.12.2021 to Bitpana
1 Address shows me 3 transfers -  1. End of staking period 1 - 2. Assembly staking round 2 -  3. Voting

But there is one Adress missing and I found it on thetangle.org 
How can I import these Adress to my firefly?

**Reply by MiKeF05** [reply]
Are you sure that the missing address belongs to this wallet? Do you use a ledger device?

**Reply by Mathias2679** [reply]
Hey, yes I'm absoluty sure!!! No, i use only the firefly Wallet. 
I can see the right addresses in the transaction overview in bitpanda. Every address from the end of 2021 until the end of the first staking periode is not in the address history.

**Reply by MiKeF05** [reply]
Which version of Firefly do you use now? How many times have you tried to run the balance finder without closing the popup?

**Reply by Mathias2679** [reply]
It's the latest version-> 1.6.2
I tried it many times - Result:
Wallets searched: 21
Addresses 300
Wallets found 1

What happen with the balance finder? Is the balance finder looking for IOTA-tokens? The balance of IOTA-Tokens are correct. I'm just missing the shimmer and assembly-token from the staking period 1

**Reply by MiKeF05** [reply]
The balance finder searches for addresses. And right now the shimmer and assembly-token are connected to your IOTA addresses. Would it be possible to join the IOTA discord server? I´m confused because you say you have 2 wallets in Firefly but only 1 wallet is found. Do you mean 2 profiles? You can send me a DM on discord (Mike_F#0572)

**Reply by MiKeF05** [reply]
Solved!

**Reply by Mathias2679** [reply]
Yes it is! Thank you so much MiKe! 💯

**Comment by Pulseicans** [comment]
have same issue missing asmb after last staking

**Comment by NICKLECITYKICKS** [comment]
I also wanted to partake in the second or most recent staking but when i used my 24 word recovery to get back in. all my balances went to zero.

**Reply by begonaalvarezd** [reply]
Have you tried to use the balance finder in the settings a few times in a row? When you restore from mnemonic, you might need to run the balance finder multiple times in a row so firefly can find all addresses, and in that process, find the ones you used for the staking events

**Reply by NICKLECITYKICKS** [reply]
i ran it up to a 50 times just a second ago and it finds one wallet but no balance : ( un like the others in the comments im missing all three tokens

**Reply by begonaalvarezd** [reply]
Just to make sure, are you connected to mainnet (your address starts with iota1..., if it starts with atoi1... your are in devnet), and if so, are you connected to the offical nodes or do you use a private one? You can check the nodes in settings, and the network in the "network icon" you have in the sidebar, if you are connected to devnet it will show it 
![image](https://user-images.githubusercontent.com/3624944/220852861-861ee0af-10f0-4f4c-9632-7affbfa68f94.png)

In case you are in devnet, please create a new profile using your 24 words but **do not select a developer profile**

**Reply by nomad-angel** [reply]
I updated Firefly to ver 2.0.3 from ver 1.5. my ASMB balance is/has gone missing. In the previous version the balance was showing during/after the staking period ended. i have since installed Firefly shimmer which also shows 0 balance. I have checked windows HOSTs File and found no entries that would block GitHub. I am connected to mainnet and not devnet. I have created a couple of profiles using my 24 words, in all cases im getting a zero balance. Im not sure what to do now, can anyone assist please ?

**Reply by begonaalvarezd** [reply]
Quick information about ASMB and SMR:

- Assembly: please read https://blog.iota.org/stardust-upgrade-iota-tokenomics/. The assembly project does not exist anymore, but your rewards got airdropped to you in form of IOTAs. They will appear in the new version 2.0.3 in the vesting tab. If you are missing balances you can run the balance finder from settings > advanced > balance finder, or if you are missing rewards you can also run the rewards finder from the vesting tab, the '....' menu, search vesting rewards
- Shimmer: what the old firefly version displayed was just a representation of what you owned, but you couldnt control the shimmers from the iota app because your rewards are in the shimmer network, not the iota network. You must download the firefly shimmer app to control those funds, you can read more about it here https://blog.shimmer.network/how-to-claim-smr-firefly/

The new firefly app is not intended to display old staking rewards, but the new iota airdrop is on the new app and the shimmer tokens should be managed from the shimmer app.

**Comment by handsomejoey** [comment]
i have the same issue both shimmer and assmbly tokens are gone

**Reply by begonaalvarezd** [reply]
Please read https://github.com/iotaledger/firefly/discussions/2972#discussioncomment-7578484
